source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -514,9 +514,18 @@ ValidationResults ContextImpl::customVerifyCertChain(
   }
   ASSERT(cert_validator_);
   const char* host_name = SSL_get_servername(ssl, TLSEXT_NAMETYPE_host_name);
+
+  CertValidator::ExtraValidationContext validation_ctx;
+  auto callbacks =
+      static_cast<Network::TransportSocketCallbacks*>(SSL_get_ex_data(ssl, sslSocketIndex()));
+  if (callbacks != nullptr) {
```

Comment:
IIRC static_cast won't return nullptr upon failure. Maybe use dynamic_cast?

Reply:
dynamic_cast for non-polymorphic types are not allowed. `SSL_get_ex_data` returns void pointer. So dynamic_cast can not be used here.

Reply:
I see. In which case can callbacks be nullptr?

Reply:
Here is how we get callback:
```
void *SSL_get_ex_data(const SSL *ssl, int idx) {
  return CRYPTO_get_ex_data(&ssl->ex_data, idx);
}
```

Based on boringssl implementation, it could be null [here](https://github.com/google/boringssl/blob/e4acd6cb568214b1c7db4e59ce54ea2e1deae1f5/crypto/ex_data.c#L200)